sideline-flymake.el

```diff
@@ -61,9 +66,15 @@ Argument COMMAND is required in sideline backend."
 
 (defun sideline-flymake--get-errors ()
   "Return flymake errors."
-  ;; Don't need to take care of the region, since sideline cannot display with
-  ;; region is active.
-  (flymake-diagnostics (point)))
+  (let ((beg (if sideline-flymake-display-errors-whole-line
+		 (line-beginning-position)
+	       (point)))
+	(end (if sideline-flymake-display-errors-whole-line
+		 (line-end-position)
+	       nil)))
+    ;; Don't need to take care of the region, since sideline cannot display with
+    ;; region is active.
+    (flymake-diagnostics beg end)))
```

Comment:
Maybe something like this? I think it's a bit cleaner.

```elisp
(cl-case sideline-flymake-display-mode
  (`point (flymake-diagnostics (point))
  (`line (flymake-diagnostics (line-beginning-position) (line-end-position))))
```

Reply:
I made those changes, but what if the user sets the variable with other symbol that's not line or point? It's fine to just ignore it?

Reply:
Good catch. I think prompt an user-error should be good? Invalid symbol..., etc.

Reply:
I made it trow a user-error, but do you know if there is a better type of error to trow?

Here is how it looks:
![image](https://user-images.githubusercontent.com/14265358/202868974-81977cb1-25b9-4a58-baf2-a709fe3e8f6d.png)

Reply:
Not as I am aware of, I think, ultimately, people would know how to configure it correctly, so I'm not too worried about the error message, but I don't want to make this type of error too aggressive.